### PR TITLE
[tests] Move Mono.Android-Tests to top tests dir

### PR DESCRIPTION
**Pull request by pjcollins**
Context: https://github.com/xamarin/xamarin-android/pull/5250

PR #5250 will be adding another another Mono.Android-Tests project file,
and it feels like a good time to move all relevant Mono.Android test
source into a single location and reduce duplication where possible.

The `src/Mono.Android/Test/` folder has been moved into the top level
`tests` directory.  The two existing `tests/Runtime-*` folders have been
moved under the new `tests/Mono.Android-Tests` folder.  Some duplication
that existed in the .targets and .projitems files that belonged to these
test projects has been cleaned up.

The obsolete `TestInstrumentation` class has been deleted in favor of
the newer `NUnitInstrumentation` class.

The `Build.GetSerial ()` API is restricted on API 29+, so a check has
been added to prevent the following exception:

    AndroidRuntime: FATAL EXCEPTION: Instr: xamarin.android.runtimetests.NUnitInstrumentation
    AndroidRuntime: Process: Mono.Android_Tests, PID: 3489
    AndroidRuntime: java.lang.SecurityException: getSerial: The user 10106 does not meet the requirements to access device identifiers.
    AndroidRuntime: 	at android.os.Parcel.createException(Parcel.java:2071)
    AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:2039)
    AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:1987)
    AndroidRuntime: 	at android.os.IDeviceIdentifiersPolicyService$Stub$Proxy.getSerialForPackage(IDeviceIdentifiersPolicyService.java:159)
    AndroidRuntime: 	at android.os.Build.getSerial(Build.java:162)

### CODE REVIEW

File: tests/Mono.Android-Tests/Runtime-AppBundle/Mono.Android-TestsAppBundle.csproj [hunk 1]
```diff
@@ -65,76 +65,88 @@
...
       <Name>TestRunner.NUnit</Name>
     </ProjectReference>
-    <ProjectReference Include="$(_MonoAndroidTest)Java.Interop-Tests\Java.Interop-Tests.csproj">
+    <ProjectReference Include="..\Java.Interop-Tests\Java.Interop-Tests.csproj">
       <Project>{6CB00820-A66B-43E5-8785-ED456C6E9F39}</Project>
       <Name>Java.Interop-Tests</Name>
     </ProjectReference>
-    <ProjectReference Include="$(_MonoAndroidTest)Mono.Android-Test.Library\Mono.Android-Test.Library.csproj">
+    <ProjectReference Include="..\Mono.Android-Test.Library\Mono.Android-Test.Library.csproj">
       <Project>{8CB5FF58-FF95-43B9-9064-9ACE9525866F}</Project>
```
Comment (jonathanpeppers) [inline tests/Mono.Android-Tests/Runtime-AppBundle/Mono.Android-TestsAppBundle.csproj:132]: Is there a few more things that could move into `Mono.Android-Test.Shared.projitems` or `Directory.Build.targets` so we have less duplication?

I'm thinking the `@(ProjectReference)`, `@(AndroidResource)`, `@(AndroidAssets)`, and `@(AndroidBoundLayout)` items would all be nearly the same.

Comment (pjcollins) [reply tests/Mono.Android-Tests/Runtime-AppBundle/Mono.Android-TestsAppBundle.csproj]: Great idea, I moved these over to `Mono.Android-Test.Shared.projitems`. I also removed the duplicate resources that belonged to `Runtime-MultiDex` and all three projects now use the one shared set of resources and assets.

File: tests/Mono.Android-Tests/Directory.Build.targets [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      <Package>$(_MonoAndroidTestPackage)</Package>
+    </TestApkPermission>
+  </ItemGroup>
+
+  <ItemGroup>
+    <AndroidNativeLibrary Include="$(MSBuildThisFileDirectory)libs\arm64-v8a\libreuse-threads.so" />
+    <AndroidNativeLibrary Include="$(MSBuildThisFileDirectory)libs\armeabi-v7a\libreuse-threads.so" />
+    <AndroidNativeLibrary Include="$(MSBuildThisFileDirectory)libs\x86\libreuse-threads.so" />
+    <AndroidNativeLibrary Include="$(MSBuildThisFileDirectory)libs\x86_64\libreuse-threads.so" />
+  </ItemGroup>
```
Comment (jonathanpeppers) [inline tests/Mono.Android-Tests/Directory.Build.targets:23]: Conceptually, would this item group make more sense to go in `Mono.Android-Test.Shared.projitems`?

**Review by jonathanpeppers — APPROVED** [review]
Otherwise, if the tests end up green, LTGM 👍

File: tests/Mono.Android-Tests/Directory.Build.targets (hunk 2 above)
Comment (pjcollins) [reply tests/Mono.Android-Tests/Directory.Build.targets]: I considered renaming this to `_NativeLibBuildOutput` and then adding the following to the shared .projitems:
```xml
<AndroidNativeLibrary Include="@(_NativeLibBuildOutput)" />
```
but decided to just move them instead.

**Review by radekdoulik — APPROVED** [review]

**Pull request merged**